internal/contour/cluster.go

```diff
@@ -97,13 +97,7 @@ func visitClusters(root dag.Vertex) map[string]*v2.Cluster {
 
 func (v *clusterVisitor) visit(vertex dag.Vertex) {
 	switch service := vertex.(type) {
-	case *dag.HTTPService:
-		name := envoy.Clustername(&service.TCPService)
-		if _, ok := v.clusters[name]; !ok {
-			c := envoy.Cluster(service)
-			v.clusters[c.Name] = c
-		}
-	case *dag.TCPService:
+	case dag.Service:
```

Comment:
This is good cleanup, but unrelated to this change. Can you please send it as its own PR. Thank you.

Reply:
It wasn't completely unrelated to this PR, I had to change `Clustername` to take a `Service` as a parameter since the hash needed to include [fields that are only present in an `HTTPService`](https://github.com/heptio/contour/pull/852/commits/b7be44985e81059703fd5c7f003a872c820d3b7a#diff-0f385e580f0bf79e437d3ad4d0b52f20R134).

The cleanup here was just a nice byproduct of changing that function, but I can still split that part out into a separate PR if you'd like.